subxt/src/metadata.rs

```diff
@@ -127,6 +137,32 @@ impl Metadata {
     pub fn runtime_metadata(&self) -> &RuntimeMetadataLastVersion {
         &self.metadata
     }
+
+    /// Obtain the pallet unique identifier.
+    pub fn pallet_hash(&self, name: &'static str) -> Result<[u8; 32], MetadataError> {
+        if let Some(cache) = self.cache.lock().unwrap().pallet_hashes.get(name) {
+            return Ok(*cache)
+        }
+
+        let metadata = self.runtime_metadata();
+        let pallet = match metadata.pallets.iter().find(|pallet| pallet.name == name) {
+            Some(pallet) => pallet,
+            _ => return Err(MetadataError::PalletNotFound(name.to_string())),
+        };
+
+        let hash = subxt_metadata::get_pallet_hash(&metadata.types, pallet);
+        let mut cache = self.cache.lock().unwrap();
+        cache.pallet_hashes.insert(name.to_string(), hash);
+        Ok(hash)
+    }
```

Comment:
Do you think this might be better off in `PalletMetadata`? That way, the hash cache is just a `Mutex<[u8; 32]>` or something, and then either way we'll do a map lookup to get hold of the `PalletMetadata`.

And then API wise, instead of doing `metadata.pallet_hash("MyPallet")` to retrieve this hash, you'd end up doing `metadata.pallet("MyPallet").hash()`.

Reply:
On chatting with you, I can see why the function is where it is now, and also why the `Arc` is there! I think we can tidy the Arc's a bit but let's do that another time :)

So, we can ignore this comment!